node/src/client/mod.rs

```diff
@@ -181,6 +204,80 @@ impl<N: Network, C: ConsensusStorage<N>> Client<N, C> {
         }));
     }
 
+    /// Initializes transaction verification.
+    fn initialize_transaction_verification(&self) {
+        // Start the transaction verification loop.
+        let node = self.clone();
+        self.handles.lock().push(tokio::spawn(async move {
+            loop {
+                // If the Ctrl-C handler registered the signal, stop the node.
+                if node.shutdown.load(Relaxed) {
+                    info!("Shutting down transaction verification");
+                    break;
+                }
+
+                // Determine the current length of the queue and counter.
+                let queue_len = node.transaction_queue.lock().len();
+                let counter = node.verification_counter.load(Relaxed);
+
+                // Sleep to allow the queue to be filled and transactions to be validated.
+                match (queue_len, counter) {
+                    // If the queue is empty, sleep longer and retry.
+                    (0, _) => {
+                        sleep(Duration::from_millis(500)).await;
```

Comment:
I'm wondering if this isn't a bit much; we probably want to eagerly verify transactions as they come

Reply:
f6debd6f4